src/client/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -129,6 +129,31 @@ public async Task<IAccount> GetAccountAsync(string accountId)
             return accounts.FirstOrDefault(account => account.HomeAccountId.Identifier.Equals(accountId, StringComparison.OrdinalIgnoreCase));
         }
 
+        /// <summary>
+        /// Get the <see cref="IAccount"/> by its identifier among the accounts available in the token cache,
+        /// based on the user flow. This is for Azure AD B2C scenarios.
+        /// </summary>
+        /// <param name="userFlow">User flow identifier. The identifier is the user flow being targeted by the specific B2C authority/>.
+        /// </param>
+        public async Task<IAccount> GetAccountByUserFlowAsync(string userFlow)
+        {
+            if (string.IsNullOrWhiteSpace(userFlow))
+            {
+                throw new ArgumentException($"{nameof(userFlow)} should not be null or whitespace", nameof(userFlow));
+            }
+
+            var accounts = await GetAccountsAsync().ConfigureAwait(false);
+
+            foreach (var account in accounts)
+            {
+                string accountIdentifier = account.HomeAccountId.ObjectId.Split('.')[0];
+                if (accountIdentifier.EndsWith(userFlow.ToLowerInvariant()))
+                    return account;
```

Comment:
A few more comments: 

- Is there only 1 account per userflow? Can't we have multiple?
- You can use `.EndsWith("stuff", System.StringComparison.OrdinalIgnoreCase);` for a more robust comparison. 
- nit, but it's cleaner to use Linq: 

```csharp
return accounts.Where(acc => acc.HomeAccountId.ObjectId.Split('.').EndsWith(...));
```

Reply:
addressed this and moved to an extension method.